src/evaluators/IfStatement.js

```diff
@@ -30,6 +31,7 @@ export function evaluate(
   let exprRef = env.evaluate(ast.test, strictCode);
   // 2. Let exprValue be ToBoolean(? GetValue(exprRef))
   let exprValue = GetValue(realm, exprRef);
+  if (exprValue instanceof AbstractValue) exprValue = simplifyAbstractValue(realm, exprValue);
```

Comment:
Is there an unwritten rule now that all calls to `GetValue` should be followed by a call to `simplifyAbstractValue`?

Reply:
This commit belongs to a previous pull request. Good question. Right now, things happen here because I did not want to add simplifier.js to the long cycle. I'm not happy about this and have some ideas of how to refactor things a bit more nicely. I'll add some relevant todo comments in a future pull request.